### ISSUE TITLE
[heading-has-content] fails when Vue component is being used inside

### ISSUE THREAD

**Issue by equinox7** [vue-a11y/eslint-plugin-vuejs-accessibility#66]
In our project we translate website into multiple languages and we use a special Vue component instead of a text node inside heading tags. Example:

```html
<h1>
  <Trans tag="hello_world">
</h1>
```

which in English renders as:

```html
<h1>
  Hello world!
</h1>
```

The plugin complains about it:
> error  Headings must have content and the content must be accessible by a screen reader  vuejs-accessibility/heading-has-content

Is there a way to implement some exception for specifying which components are allowed to be used inside headings?

**Comment by vhoyer** [comment]
This indeed seems like an oversight from the lint rule :thinking:

Could you open a PR for this? We could add an array of allowed components.

**Comment by rchl** [comment]
I'd like to sneak in what I think is a relevant issue:

In my case, I'm using `v-t` directive on the element which has the same purpose - to set the element's text to the translated string. This also complains while it shouldn't.

**Comment by vhoyer** [comment]
In that case we should check the `v-html` too

**Comment by rchl** [comment]
And `v-text`.
But while it would make sense to have `v-html` and `v-text` included by default, I guess other should be an option on the rule.

**Comment by vhoyer** [comment]
That also crossed my mind, thanks for mentioning haha

**Comment by kddeisz** [comment]
Those directives are already included (`v-text` and `v-html`) https://github.com/vue-a11y/eslint-plugin-vuejs-accessibility/blob/master/src/utils/hasContent.js#L27-L28

**Comment by kddeisz** [comment]
I think it makes sense to have exceptions here though, I can bake that in
